mantidproject/mantid
# Refactor enginx script to call UI code

**Description of work.**

Rewrite EnginX script to call GUI code so 

1. Have system test coverage for GUI code
2. Can be used in autoreduction (and will give same answer as GUI)

A couple of changes to an underlying algorithm PDCalibration:

- Fixed bug in PDCalibration such that it now correctly minimises chi-squared (cause of #33046) 
- Added ability to specify fit range for each peak rather than a global window in d-spacing.

**To test:**

(1) Run the systems test script on a non-windows machine to check it passes OK
`systemtest.bat -C debug -R EnginXScriptTest -j 2`

Fixes #32663 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
